src/features/ValueEditor/ValueEditor.tsx

```diff
@@ -64,7 +64,10 @@ export const Validation = ({schema, value, index, onChange, onAdd, onRemove, onC
   const validation = schema
     ? zod.validateValue(schema, value)
     : {success: false as const, error: 'Invalid schema'}
-  const parsedData = validation.success && JSON.stringify(validation.data)
+
+  const parsedData =
```

Comment:
We could have also developed a patch for BigInt's toJSON() method. As MDN points out, this is one of the rare cases where this practice is not discouraged.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json

Anyway, I think your implementation is good too. I leave the decision to you